File: mongo-examples/src/main/java/io/vertx/examples/mongo/MongoClientVerticle.java
package io.vertx.examples.mongo;

import io.vertx.core.Future;
import io.vertx.core.VerticleBase;
import io.vertx.core.Vertx;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.mongo.MongoClient;
import io.vertx.launcher.application.VertxApplication;

public class MongoClientVerticle extends VerticleBase {

  public static void main(String[] args) {
    VertxApplication.main(new String[]{MongoClientVerticle.class.getName()});
  }

  private MongoClient mongoClient;

  @Override
  public Future<?> start() throws Exception {

    JsonObject config = Vertx.currentContext().config();

    String uri = config.getString("mongo_uri");
    if (uri == null) {
      uri = "mongodb://localhost:27017";
    }
    String db = config.getString("mongo_db");
    if (db == null) {
      db = "test";
    }

    JsonObject mongoconfig = new JsonObject()
      .put("connection_string", uri)
      .put("db_name", db);

    mongoClient = MongoClient.createShared(vertx, mongoconfig);

    JsonObject product1 = new JsonObject().put("itemId", "12345").put("name", "Cooler").put("price", "100.0");

    return mongoClient.save("products", product1)
      .compose(id -> {
        System.out.println("Inserted id: " + id);
        return mongoClient.find("products", new JsonObject().put("itemId", "12345"));
      })
      .compose(res -> {
        System.out.println("Name is " + res.get(0).getString("name"));
        return mongoClient.removeDocument("products", new JsonObject().put("itemId", "12345"));
      })
      .onSuccess(res -> {
        System.out.println("Product removed ");
      });
  }
}

File: mongo-examples/src/main/kotlin/io/vertx/examples/mongo/MongoClientVerticle.kt
package io.vertx.examples.mongo

import io.vertx.core.Vertx
import io.vertx.ext.mongo.MongoClient
import io.vertx.kotlin.core.json.*

class MongoClientVerticle : io.vertx.core.AbstractVerticle()  {
  override fun start() {

    var config = Vertx.currentContext().config()

    var uri = config.getString("mongo_uri")
    if (uri == null) {
      uri = "mongodb://localhost:27017"
    }
    var db = config.getString("mongo_db")
    if (db == null) {
      db = "test"
    }

    var mongoconfig = json {
      obj(
        "connection_string" to uri,
        "db_name" to db
      )
    }

    var mongoClient = MongoClient.createShared(vertx, mongoconfig)

    var product1 = json {
      obj(
        "itemId" to "12345",
        "name" to "Cooler",
        "price" to "100.0"
      )
    }

    mongoClient.save("products", product1, { id ->
      println("Inserted id: ${id.result()}")

      mongoClient.find("products", json {
        obj("itemId" to "12345")
      }, { res ->
        println("Name is ${res.result()[0].getString("name")}")

        mongoClient.remove("products", json {
          obj("itemId" to "12345")
        }, { rs ->
          if (rs.succeeded()) {
            println("Product removed ")
          }
        })

      })

    })

  }
}
